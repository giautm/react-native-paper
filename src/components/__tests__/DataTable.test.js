/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import DataTable from '../DataTable/DataTable';

it('renders data table head', () => {
  const tree = renderer
    .create(
      <DataTable.Head>
        <DataTable.HeaderCell>Dessert</DataTable.HeaderCell>
        <DataTable.HeaderCell>Calories</DataTable.HeaderCell>
      </DataTable.Head>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders data table header cell', () => {
  const tree = renderer
    .create(
      <DataTable.HeaderCell size={4} sortDirection="descending">
        Dessert
      </DataTable.HeaderCell>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders numeric data table header cell', () => {
  const tree = renderer
    .create(<DataTable.HeaderCell numeric>Calories</DataTable.HeaderCell>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders data table header cell with press handler', () => {
  const tree = renderer
    .create(
      <DataTable.HeaderCell
        size={4}
        sortDirection="descending"
        press={() => {}}
      >
        Dessert
      </DataTable.HeaderCell>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders data table body', () => {
  const tree = renderer
    .create(
      <DataTable.Body>
        <DataTable.Row>
          <DataTable.Cell>Cupcake</DataTable.Cell>
          <DataTable.Cell>356</DataTable.Cell>
        </DataTable.Row>
        <DataTable.Row>
          <DataTable.Cell>Eclair</DataTable.Cell>
          <DataTable.Cell>262</DataTable.Cell>
        </DataTable.Row>
      </DataTable.Body>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders data table cell', () => {
  const tree = renderer
    .create(<DataTable.Cell size={4}>Cupcake</DataTable.Cell>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders numeric data table cell', () => {
  const tree = renderer
    .create(<DataTable.Cell numeric>356</DataTable.Cell>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders data table pagination', () => {
  const tree = renderer
    .create(
      <DataTable.Pagination
        count={15}
        page={3}
        rowsPerPage={4}
        onChangePage={() => {}}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders data table pagination with custom text', () => {
  const tree = renderer
    .create(
      <DataTable.Pagination
        count={15}
        page={3}
        rowsPerPage={4}
        onChangePage={() => {}}
        labelDisplayedRows={({ from, to, count }) =>
          `${from}-${to} von ${count}`
        }
        labelRowsPerPage="Zeilen pro Seite:"
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
