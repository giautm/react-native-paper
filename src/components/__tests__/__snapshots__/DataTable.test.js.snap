// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders data table body 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#f5f5f5",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "flexDirection": "row",
            "height": 48,
            "paddingHorizontal": 16,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                },
                Array [
                  Object {
                    "color": "#000000DE",
                    "fontSize": 14,
                  },
                  Object {
                    "flexDirection": "row",
                    "textAlign": "left",
                  },
                  undefined,
                ],
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "accent": "#03dac4",
                  "background": "#f6f6f6",
                  "disabled": "rgba(0, 0, 0, 0.26)",
                  "error": "#B00020",
                  "placeholder": "rgba(0, 0, 0, 0.38)",
                  "primary": "#6200ee",
                  "surface": "#ffffff",
                  "text": "#000000",
                },
                "dark": false,
                "fonts": Object {
                  "light": "HelveticaNeue-Light",
                  "medium": "HelveticaNeue-Medium",
                  "regular": "Helvetica Neue",
                  "thin": "HelveticaNeue-Thin",
                },
                "roundness": 4,
              }
            }
          >
            Cupcake
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                },
                Array [
                  Object {
                    "color": "#000000DE",
                    "fontSize": 14,
                  },
                  Object {
                    "flexDirection": "row",
                    "textAlign": "left",
                  },
                  undefined,
                ],
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "accent": "#03dac4",
                  "background": "#f6f6f6",
                  "disabled": "rgba(0, 0, 0, 0.26)",
                  "error": "#B00020",
                  "placeholder": "rgba(0, 0, 0, 0.38)",
                  "primary": "#6200ee",
                  "surface": "#ffffff",
                  "text": "#000000",
                },
                "dark": false,
                "fonts": Object {
                  "light": "HelveticaNeue-Light",
                  "medium": "HelveticaNeue-Medium",
                  "regular": "Helvetica Neue",
                  "thin": "HelveticaNeue-Thin",
                },
                "roundness": 4,
              }
            }
          >
            356
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#f5f5f5",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "flexDirection": "row",
            "height": 48,
            "paddingHorizontal": 16,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                },
                Array [
                  Object {
                    "color": "#000000DE",
                    "fontSize": 14,
                  },
                  Object {
                    "flexDirection": "row",
                    "textAlign": "left",
                  },
                  undefined,
                ],
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "accent": "#03dac4",
                  "background": "#f6f6f6",
                  "disabled": "rgba(0, 0, 0, 0.26)",
                  "error": "#B00020",
                  "placeholder": "rgba(0, 0, 0, 0.38)",
                  "primary": "#6200ee",
                  "surface": "#ffffff",
                  "text": "#000000",
                },
                "dark": false,
                "fonts": Object {
                  "light": "HelveticaNeue-Light",
                  "medium": "HelveticaNeue-Medium",
                  "regular": "Helvetica Neue",
                  "thin": "HelveticaNeue-Thin",
                },
                "roundness": 4,
              }
            }
          >
            Eclair
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                },
                Array [
                  Object {
                    "color": "#000000DE",
                    "fontSize": 14,
                  },
                  Object {
                    "flexDirection": "row",
                    "textAlign": "left",
                  },
                  undefined,
                ],
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "accent": "#03dac4",
                  "background": "#f6f6f6",
                  "disabled": "rgba(0, 0, 0, 0.26)",
                  "error": "#B00020",
                  "placeholder": "rgba(0, 0, 0, 0.38)",
                  "primary": "#6200ee",
                  "surface": "#ffffff",
                  "text": "#000000",
                },
                "dark": false,
                "fonts": Object {
                  "light": "HelveticaNeue-Light",
                  "medium": "HelveticaNeue-Medium",
                  "regular": "Helvetica Neue",
                  "thin": "HelveticaNeue-Thin",
                },
                "roundness": 4,
              }
            }
          >
            262
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders data table cell 1`] = `
<View
  style={
    Object {
      "flex": 4,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
          },
          Array [
            Object {
              "color": "#000000DE",
              "fontSize": 14,
            },
            Object {
              "flexDirection": "row",
              "textAlign": "left",
            },
            undefined,
          ],
        ]
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "placeholder": "rgba(0, 0, 0, 0.38)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "HelveticaNeue-Light",
            "medium": "HelveticaNeue-Medium",
            "regular": "Helvetica Neue",
            "thin": "HelveticaNeue-Thin",
          },
          "roundness": 4,
        }
      }
    >
      Cupcake
    </Text>
  </View>
</View>
`;

exports[`renders data table head 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 48,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "borderBottomColor": "#f5f5f5",
              "borderBottomWidth": 1,
              "borderStyle": "solid",
              "flexDirection": "row",
              "paddingVertical": 12,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#00000099",
                  "flex": 1,
                  "fontSize": 12,
                  "fontWeight": "500",
                  "height": 24,
                  "lineHeight": 24,
                },
                Object {
                  "textAlign": "left",
                },
                null,
                undefined,
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#03dac4",
                "background": "#f6f6f6",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "error": "#B00020",
                "placeholder": "rgba(0, 0, 0, 0.38)",
                "primary": "#6200ee",
                "surface": "#ffffff",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 4,
            }
          }
        >
          Dessert
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "borderBottomColor": "#f5f5f5",
              "borderBottomWidth": 1,
              "borderStyle": "solid",
              "flexDirection": "row",
              "paddingVertical": 12,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#00000099",
                  "flex": 1,
                  "fontSize": 12,
                  "fontWeight": "500",
                  "height": 24,
                  "lineHeight": 24,
                },
                Object {
                  "textAlign": "left",
                },
                null,
                undefined,
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#03dac4",
                "background": "#f6f6f6",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "error": "#B00020",
                "placeholder": "rgba(0, 0, 0, 0.38)",
                "primary": "#6200ee",
                "surface": "#ffffff",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 4,
            }
          }
        >
          Calories
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 0.5,
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;

exports[`renders data table header cell 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 4,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "borderBottomColor": "#f5f5f5",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "flexDirection": "row",
          "paddingVertical": 12,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "height": 24,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "rotate": "180deg",
            },
          ],
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 16,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
          },
          Array [
            Object {
              "alignItems": "center",
              "color": "#00000099",
              "flex": 1,
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            undefined,
          ],
        ]
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "placeholder": "rgba(0, 0, 0, 0.38)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "HelveticaNeue-Light",
            "medium": "HelveticaNeue-Medium",
            "regular": "Helvetica Neue",
            "thin": "HelveticaNeue-Thin",
          },
          "roundness": 4,
        }
      }
    >
      Dessert
    </Text>
  </View>
</View>
`;

exports[`renders data table header cell with press handler 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 4,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "borderBottomColor": "#f5f5f5",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "flexDirection": "row",
          "paddingVertical": 12,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "height": 24,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "rotate": "180deg",
            },
          ],
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 16,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
          },
          Array [
            Object {
              "alignItems": "center",
              "color": "#00000099",
              "flex": 1,
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            undefined,
          ],
        ]
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "placeholder": "rgba(0, 0, 0, 0.38)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "HelveticaNeue-Light",
            "medium": "HelveticaNeue-Medium",
            "regular": "Helvetica Neue",
            "thin": "HelveticaNeue-Thin",
          },
          "roundness": 4,
        }
      }
    >
      Dessert
    </Text>
  </View>
</View>
`;

exports[`renders data table pagination 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingLeft": 16,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
        },
        Array [
          Object {
            "color": "#00000099",
            "fontSize": 12,
          },
          Object {
            "marginRight": 30,
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#03dac4",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "placeholder": "rgba(0, 0, 0, 0.38)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 4,
      }
    }
  >
    Rows per page:
     
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
        },
        Array [
          Object {
            "color": "#00000099",
            "fontSize": 12,
          },
          Object {
            "marginLeft": 30,
            "marginRight": 44,
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#03dac4",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "placeholder": "rgba(0, 0, 0, 0.38)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 4,
      }
    }
  >
    13-15 of 15
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits="button"
    accessible={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 18,
          "height": 36,
          "justifyContent": "center",
          "margin": 6,
          "width": 36,
        },
        false,
        undefined,
      ]
    }
  >
    <View
      style={null}
    >
      <Text
        accessibilityElementsHidden={true}
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 18,
          "height": 36,
          "justifyContent": "center",
          "margin": 6,
          "width": 36,
        },
        Object {
          "opacity": 0.32,
        },
        undefined,
      ]
    }
  >
    <View
      style={null}
    >
      <Text
        accessibilityElementsHidden={true}
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders data table pagination with custom text 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingLeft": 16,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
        },
        Array [
          Object {
            "color": "#00000099",
            "fontSize": 12,
          },
          Object {
            "marginRight": 30,
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#03dac4",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "placeholder": "rgba(0, 0, 0, 0.38)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 4,
      }
    }
  >
    Zeilen pro Seite:
     
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
        },
        Array [
          Object {
            "color": "#00000099",
            "fontSize": 12,
          },
          Object {
            "marginLeft": 30,
            "marginRight": 44,
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#03dac4",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "placeholder": "rgba(0, 0, 0, 0.38)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 4,
      }
    }
  >
    13-15 von 15
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits="button"
    accessible={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 18,
          "height": 36,
          "justifyContent": "center",
          "margin": 6,
          "width": 36,
        },
        false,
        undefined,
      ]
    }
  >
    <View
      style={null}
    >
      <Text
        accessibilityElementsHidden={true}
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 18,
          "height": 36,
          "justifyContent": "center",
          "margin": 6,
          "width": 36,
        },
        Object {
          "opacity": 0.32,
        },
        undefined,
      ]
    }
  >
    <View
      style={null}
    >
      <Text
        accessibilityElementsHidden={true}
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders numeric data table cell 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
          },
          Array [
            Object {
              "color": "#000000DE",
              "fontSize": 14,
            },
            Object {
              "flexDirection": "row-reverse",
              "textAlign": "right",
            },
            undefined,
          ],
        ]
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "placeholder": "rgba(0, 0, 0, 0.38)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "HelveticaNeue-Light",
            "medium": "HelveticaNeue-Medium",
            "regular": "Helvetica Neue",
            "thin": "HelveticaNeue-Thin",
          },
          "roundness": 4,
        }
      }
    >
      356
    </Text>
  </View>
</View>
`;

exports[`renders numeric data table header cell 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "borderBottomColor": "#f5f5f5",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "flexDirection": "row",
          "paddingVertical": 12,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
          },
          Array [
            Object {
              "alignItems": "center",
              "color": "#00000099",
              "flex": 1,
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "textAlign": "right",
            },
            null,
            undefined,
          ],
        ]
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "placeholder": "rgba(0, 0, 0, 0.38)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "HelveticaNeue-Light",
            "medium": "HelveticaNeue-Medium",
            "regular": "Helvetica Neue",
            "thin": "HelveticaNeue-Thin",
          },
          "roundness": 4,
        }
      }
    >
      Calories
    </Text>
  </View>
</View>
`;
